org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -159,18 +163,37 @@ private void updateReplacement(CompletionProposal proposal, CompletionItem item,
 			}
 		}
 
-		if (range == null) {
-			boolean completionOverwrite = preferences.isCompletionOverwrite();
-			if (!completionOverwrite && (proposal.getKind() == CompletionProposal.METHOD_REF || proposal.getKind() == CompletionProposal.LOCAL_VARIABLE_REF || proposal.getKind() == CompletionProposal.FIELD_REF)) {
```

Comment:
@snjeza Do you remember why here we only change the replace range for these completion proposals in the insert mode?

Reply:
@jdneo If the replace range is already defined, we shouldn't change it., I suppose.

Reply:
I mean we only update the proposal with kind `METHOD_REF`, `LOCAL_VARIABLE_REF`, `FIELD_REF`, but also for others. Maybe we can update for other kinds as well? an example might be:

```java
new String|"some string constant";
```

when user trigger the completion after `String`(at `|`), we might eat the following constant even in the insert mode.